### PR TITLE
SISRP-22589 - Remove Berkeley::SummerSubTerm and dependencies

### PR DESCRIPTION
**Pull request by redconfetti**
https://jira.berkeley.edu/browse/SISRP-22589

I got caught up trying to fix the migrations that referenced the model itself that I was removing (Berkeley::SummerSubTerm). After a discussion with @lyttam and @nkutub we concluded that Rails migrations don't need to be fixed across the board. If anything we should perhaps flatten them to setup only the schema we currently have to clean them up.

I left the code changes I made to fix the migrations that go back to when this table was made, but not further back than that.

### CODE REVIEW

File: .rspec [hunk 1]
```diff
@@ -1,2 +1,3 @@
 --color
 --require spec_helper
+
```
Comment (davidlimcheng) [inline .rspec]: @redconfetti Erroneous newline here?

**Comment by davidlimcheng** [comment]
Whew, LGTM @redconfetti.  Just a question about a new line.

File: .rspec (hunk 1 above)
Comment (redconfetti) [reply .rspec]: @davidlimcheng - No. See [ThoughtBot - No Newline at End of File](https://robots.thoughtbot.com/no-newline-at-end-of-file) or [Stack Overflow - Why is it recommended to have empty line in the end of file](https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-file).

Our [.editorconfig](https://github.com/ets-berkeley-edu/calcentral/blob/066cba9a/.editorconfig#L7) specifies this behavior, which the plugin I use in Sublime Text enforces.

Comment (davidlimcheng) [reply .rspec]: Thanks @redconfetti! Good read.

**Pull request merged**